src/components/select/Select.js

```diff
@@ -1751,8 +1756,11 @@ export default class SelectComponent extends ListComponent {
   asString(value, options = {}) {
     value = value ?? this.getValue();
 
-    if (options.modalPreview) {
-      const template = this.itemTemplate(value, value);
+    if (options.modalPreview || this.inDataTable) {
+      if (this.inDataTable) {
+        value = this.undoValueTyping(value);
+      }
+      const template = this.itemTemplate((this.isEntireObjectDisplay() && !_.isObject(value.data)) ? { data: value } : value, value, options);
```

Comment:
Separate the function parameter value determined by ternary operator to make it easier to read.
```suggestion
     const templateValue = (this.isEntireObjectDisplay() && !_.isObject(value.data)) ? { data: value } : value;
      const template = this.itemTemplate(templateValue, value, options);
```